File: react-native-nitro-player/android/src/main/java/com/margelo/nitro/nitroplayer/playlist/Playlist.kt
package com.margelo.nitro.nitroplayer.playlist

import com.margelo.nitro.nitroplayer.TrackItem

/**
 * Represents a playlist containing multiple tracks
 * Uses ExoPlayer's native playlist functionality
 */
data class Playlist(
    val id: String,
    val name: String,
    val description: String? = null,
    val artwork: String? = null,
    val tracks: MutableList<TrackItem> = mutableListOf(),
) {
    fun toTrackItemArray(): Array<TrackItem> = tracks.toTypedArray()

    fun getTrackCount(): Int = tracks.size

    fun isEmpty(): Boolean = tracks.isEmpty()
}

File: react-native-nitro-player/ios/playlist/Playlist.swift
//
//  Playlist.swift
//  NitroPlayer
//
//  Created on 10/12/25.
//

import Foundation

/**
 * Represents a playlist containing multiple tracks
 * Uses AVPlayer's native playlist functionality
 */
class Playlist {
    let id: String
    let name: String
    let description: String?
    let artwork: String?
    var tracks: [TrackItem]
    
    init(id: String, name: String, description: String? = nil, artwork: String? = nil, tracks: [TrackItem] = []) {
        self.id = id
        self.name = name
        self.description = description
        self.artwork = artwork
        self.tracks = tracks
    }
    
    func getTrackCount() -> Int {
        return tracks.count
    }
    
    func isEmpty() -> Bool {
        return tracks.isEmpty
    }
}

